lib/puma/launcher.rb

```diff
@@ -140,7 +140,10 @@ def write_state
     # Delete the configured pidfile
     def delete_pidfile
       path = @options[:pidfile]
-      File.unlink(path) if path && File.exist?(path)
+      begin
+        File.unlink(path) if path
```

Comment:
We are still making a change to how this function works regardless of whether there are race conditions or not. So the question is if there are other errors that arise when not checking File.exist?(...) apart from the ENOENT.

> ENOENT: A component in pathname does not exist or is a dangling
> symbolic link, or pathname is empty.

Given the definition of the error after the syscall, I am assuming the behaviour is the same as the `File.exist?` but just without race conditions